builder/builder-next/exporter/writer.go

```diff
@@ -19,6 +18,7 @@ import (
 // const (
 // 	emptyGZLayer = digest.Digest("sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1")
 // )
+const imageConfigField = "moby.buildkit.buildinfo.v1"
```

Comment:
So, slightly wondering;

- As this is the master branch; does it already use the replacement? If so, perhaps we should (can we?) remove the functionality as a whole?
- If we _cannot_ remove it now, the alternative would be to add a `//nolint:...` (with a comment about the deprecation)

Advantage of the `//nolint:` would be that we deliberately make the build break once we would update BuildKit to v0.12, that way we don't accidentally forget about removing it once it's gone (whereas having a local `const` means we could still have (dead)code.

Reply:
Oh it's the master branch. Then yeah better to remove it entirely. related commit: https://github.com/moby/moby/pull/43239/commits/fda0226a890b9948a754c48fc3ad5f6e02f43f40

Reply:
Thanks! Yes, so this is master (23.1.0 "dev"), which _will_ be the first release with BuildKit 0.11, but hasn't been released yet, so perhaps it's fine to deprecate/remove ahead of BuildKit.

So my suggestion would be;


- Open a PR to mark the feature deprecated (from the "Docker Engine" perspective) in https://github.com/docker/cli/blob/master/docs/deprecated.md. We can probably mark it as "deprecated in v23.0.0", and "to be removed in v23.1.0" (as it was already deprecated in BuildKit, but we didn't mention it)
- Remove it from master
- Vendor buildkit v0.11 "latest"

Reply:
But I could use feedback on that suggestion, as I'm not very familiar with the feature (and who/how it's used) 😅